### PR TITLE
Load nested hitobjects during map load

### LINKED ISSUE #9659: long freezes at the start of spinners in catch
**Describe the bug:** 

I saw manny comments regarding this problem in the spotlights rooms, before you start a spinner freezes for about 2 seconds, 
after freezing ends, performance while spinner lasts is very low.

**Screenshots or videos showing encountered issue:** 

https://drive.google.com/file/d/1NuY64atUPQYGzws8ZaRRSvZEvHvgjKmI/view?usp=sharing

**osu!lazer version:** 

2020.723.0

**Logs:**

[performance-update.log](https://github.com/ppy/osu/files/4968050/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/4968051/runtime.log)
[updater.log](https://github.com/ppy/osu/files/4968052/updater.log)
[database.log](https://github.com/ppy/osu/files/4968053/database.log)
[network.log](https://github.com/ppy/osu/files/4968054/network.log)
[performance.log](https://github.com/ppy/osu/files/4968055/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/4968056/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/4968057/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/4968058/performance-input.log)

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes https://github.com/ppy/osu/issues/9659

1. Loads nested hitobjects, samples, etc, asynchronously.
Note that does increase memory usage even in osu!, but I think we want this anyway and save memory usage for DHO pooling.
2. Even after the above, I still had a stutter which was caused by multiple `computeInitialState`s running.
From: 
Screenshot_2020-07-24_19-17-15
To:
Screenshot_2020-07-24_19-29-57

### CODE REVIEW

File: osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs [hunk 1]
```diff
@@ -215,16 +217,18 @@ protected override void Update()
...
-                        objCache = hitObjectInitialStateCache[obj] = new Cached();
+                    if (!hitObjectInitialStateCache.TryGetValue(obj, out var state))
+                        state = hitObjectInitialStateCache[obj] = new InitialState(new Cached());
 
-                    if (objCache.IsValid)
+                    if (state.Cache.IsValid)
                         continue;
 
+                    state.ScheduledComputation?.Cancel();
+                    state.ScheduledComputation = computeInitialStateRecursive(obj);
```
Comment (peppy) [inline osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs:227]: this is really hard to follow... might help to have some inline comments explaining what this is doing

**Comment by peppy** [comment]
- In the above benchmark, are you saying you still feel a stutter from the 2.3ms remaining load?
- Have you tested this on some heavy osu! maps to make sure the async overhead hasn't negatively affected load performance?

**Comment by smoogipoo** [comment]
1. The stutter was from the 60ms, which happens the very first update frame. The 2nd screenshot is what it currently is, which is pretty unnoticeable.
2. I've tested on the Mashiro-iro Symphony Medley, and mem usage went from 1.3GB -> 1.8GB, load time felt similar-ish.

**Comment by peppy** [comment]
I'm guessing that second memory figure is what you end with at the end of the map anyway?

**Comment by smoogipoo** [comment]
You would yeah. Hard to test that though since memory usage goes up naturally - VBOs, judgements/results/etc.

**Review by peppy — APPROVED** [review]
let's just go with this for now, local enough to not be a huge issue.
